### ISSUE TITLE
How to reproduce finetuning relative depth model to metric depth mentioned in the paper?

### ISSUE THREAD

**Issue by Bearick** [DepthAnything/Depth-Anything-V2#102]
I am a postgraduate in relative fields. I think the released checkpoints are incredible, but I have trouble finetuning the model to metric depth.
I was trying to finetune the pretrained relative depth model (depth_anything_v2_vitl.pth) to metric depth in NYUv2 dataset. I use the train.py script in metric_depth and the results are horrible. I tried several data augmentation methods and adjusted lr to 1e-7/1e-8, but still struggling to get comparable results.
So I wonder how did you finetune your relative depth model to metric depth? The paper just mentioned ZoeDepth and I failed to find any solution to reproduce it.
Thanks a lot.

**Comment by Bearick** [comment]
@LiheYoung Could you provide some help? Thanks again.

**Comment by Eyvaz27** [comment]
Currently, I am trying to replace SiLogLoss with L1Loss in metric_depth/train.py while running backbone below with torch.no_grad() such that only the hierarchical depth head is affected not the underlying features (loss seems to be decreasing and also reporting metric difference to the gt depth values - also during evaluation) https://github.com/DepthAnything/Depth-Anything-V2/blob/31dc97708961675ce6b3a8d8ffa729170a4aa273/metric_depth/depth_anything_v2/dpt.py#L181

**Comment by shilpaullas97** [comment]
> I am a postgraduate in relative fields. I think the released checkpoints are incredible, but I have trouble finetuning the model to metric depth. I was trying to finetune the pretrained relative depth model (depth_anything_v2_vitl.pth) to metric depth in NYUv2 dataset. I use the train.py script in metric_depth and the results are horrible. I tried several data augmentation methods and adjusted lr to 1e-7/1e-8, but still struggling to get comparable results. So I wonder how did you finetune your relative depth model to metric depth? The paper just mentioned ZoeDepth and I failed to find any solution to reproduce it. Thanks a lot.

Hi @Bearick ,

I'm also facing a similar issue.
Did you find any way to improve the results?

**Comment by hcscysh** [comment]
@Eyvaz27 ,I want to know as you said ,train while running backbone below with torch.no_grad() such that only the hierarchical depth head is affected not the underlying features,How did you achieve it?I use the following code,        
for param in self.pretrained.parameters():
        param.requires_grad = False
but the result is worse,can you help me how to solve it ,thank you very much!

**Comment by Eyvaz27** [comment]
Heyooo, 
just setting self.pretrained,eval() on line https://github.com/DepthAnything/Depth-Anything-V2/blob/31dc97708961675ce6b3a8d8ffa729170a4aa273/metric_depth/depth_anything_v2/dpt.py#L175
should be enough. 
Note: "Results are getting worse" is a bit counter-intuitive as the DepthAnythingV2 pre-trained weights are not metric usually and when you are fitting to your dataset (as long as SiLogLoss is decreasing) you are in a good situation

**Comment by hcscysh** [comment]
@Eyvaz27,thanks for your reply.When I train the network,the loss oscillates around 0.1. Other indicators seem to be good, but the final result is not satisfactory. I want to know how to determine whether the network has converged, when the loss has reached a certain value, or when other indicators have reached a specified value. The following is my training log
![1734666576873](https://github.com/user-attachments/assets/ac0982d0-c4ef-4906-9759-eafb1bfc3f47)

**Comment by Skapis9999** [comment]
@Bearick Have you solved your issues?
